modules/xcode/_preload.lua

```diff
@@ -6,6 +6,25 @@
 
 	local p = premake
 
+--
+-- Register the ios extension
+--
+
+	local osoption = p.option.get("os")
+	if type(osoption) == "table" then
+		local ios = false
+		table.foreachi(osoption.allowed, function(allowed)
+			ios = ios or allowed[1] == "ios"
+		end)
+		if not ios then
+			table.insert(osoption.allowed, { "ios",  "iOS" })
+		end
+	end
+
+	-- add system tags for ios.
+	if not os.systemTags[p.IOS] then
+		os.systemTags[p.IOS] = { "ios", "mobile" }
```

Comment:
For both this and the `p.MACOSX` entry, it might be worth adding an `"apple"` entry? This would allow all of the checks you changed below to be `apple` instead and they'd work for tvOS and watchOS - and whatever comes next. Thoughts?

Reply:
I see. I think that it looks very good.

Reply:
I feel it is better to define `darwin` instead of `apple`.
`macosx` already has that value.

```lua
	os.systemTags =
	{
		["aix"]      = { "aix",     "posix" },
		["bsd"]      = { "bsd",     "posix" },
		["haiku"]    = { "haiku",   "posix" },
		["ios"]      = { "ios",     "darwin", "mobile" }
		["linux"]    = { "linux",   "posix" },
		["macosx"]   = { "macosx",  "darwin", "posix" },
		["solaris"]  = { "solaris", "posix" },
		["windows"]  = { "windows", "win32" },
	}
```

It may be better to include `posix`.

Reply:
I agree, perhaps both `darwin` and `posix`?